x/leverage/keeper/invariants.go

```diff
@@ -297,3 +304,39 @@ func InterestScalarsInvariant(k Keeper) sdk.Invariant {
 		), broken
 	}
 }
+
+// ExchangeRatesInvariant checks that all denoms have an uToken exchange rate >= 1
+func ExchangeRatesInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		var (
+			msg   string
+			count int
+		)
+
+		tokenPrefix := types.KeyPrefixRegisteredToken
+
+		// Iterate through all denoms of registered tokens in the
+		// keeper, ensuring none have an interest scalar less than one.
+		err := k.iterate(ctx, tokenPrefix, func(key, _ []byte) error {
+			denom := types.DenomFromKey(key, tokenPrefix)
+
+			exchangeRate := k.DeriveExchangeRate(ctx, denom)
+
+			if exchangeRate.LT(sdk.OneDec()) {
+				count++
+				msg += fmt.Sprintf("\t%s exchange rate %s is less than one\n", denom, exchangeRate.String())
+			}
+			return nil
+		})
+		if err != nil {
+			msg += fmt.Sprintf("\tSome error occurred while iterating through the uToken exchange rates %+v\n", err)
+		}
+
+		broken := count != 0
+
+		return sdk.FormatInvariant(
+			types.ModuleName, routeExchangeRates,
+			fmt.Sprintf("amount of uToken exchange rates lower than one %d\n%s", count, msg),
+		), broken
+	}
+}
```

Comment:
do we really want to check it after each block by every validator? I think we should have that function only invalidate genesis.

Reply:
Some of these would be expensive computationally to do each block, but having checked other modules (e.g. [x/bank invariants](https://github.com/cosmos/cosmos-sdk/blob/main/x/bank/keeper/invariants.go) I see that iteration over balances is done there as well.

Is there some setting that validators are using to run invariants less often as they operate other chains?

Reply:
No, there is no such setting. But we can code it. The x/bank invariants are often disabled, because they take lot of time.

Reply:
I've made a task for this issue: https://github.com/umee-network/umee/issues/1324

Reply:
BTW, the bank invariants are not even registered in the simapp.